lib/async.ts

```diff
@@ -210,22 +210,36 @@ export class Context {
     // the id is either:
     // 1. a provided string in the case of a deferred execution
     // 2. a generated string in the case of an ordinary execution
-    const id = typeof func === "string" ? func : `${parent.id}.${parent.counter}`;
+    const id = typeof func === "string" ? func : `${parent.id}.${parent.counter}.${func.name}`;
```

Comment:
This is probably the controvertial change in this PR. I added the name of the function to the id. My idea is to have more insight into the distributed call graph, once we get to observability.

Reply:
Looks good! I like that it's informational only, we don't rely on the name of the function to ensure uniqueness